File: Main/Source/GlobalSwapShop_sln/Source/GlobalSwapShop_web/FVK/ComponentParts/PropertyData.cs
// WARNING: 
// The source code of this class can be used and modified to your particular needs. 
// Sharing the class with unregistered users or using it with unregistered websites 
// or Facebook applications is not allowed and violates the rights of intellectual 
// property. 

using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace FVK
{
    /// <summary>
    /// The class contains data of Property for stream publishing
    /// </summary>
    public class PropertyData: IJSONSerialize
    {
        /// <summary>
        /// Create new undefined PropertyData object
        /// </summary>
        public PropertyData() { }

        /// <summary>
        /// Create and define new PropertyData object
        /// </summary>
        /// <param name="name">Property name</param>
        /// <param name="text">Propery text</param>
        public PropertyData(string name, string text)
        {
            this.name = name;
            this.text = text;
        }

        /// <summary>
        /// Propery name
        /// </summary>
        public string Name { get { return name; } set { name = value; } }

        /// <summary>
        /// Property text
        /// </summary>
        public string Text { get { return text; } set { text = value; } }

        /// <summary>
        /// Serialize Propery data in JSON format
        /// </summary>
        /// <returns>JSON formated string</returns>
        public string JSONSerialize()
        {
            return "\"" + name + "\":\"" + text + "\"";
        }


        /// <summary>
        /// Protected members
        /// </summary>
        protected string name;
        protected string text;
    }
}

File: Main/ReferenceDLLs/Pack1/Facebook Visual Kit VB/FVK Source Code/FVK/FVK/ComponentParts/PropertyData.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web

Namespace FVK
    ''' <summary>
    ''' The class contains data of Property for stream publishing
    ''' </summary>
    Public Class PropertyData
        Implements IJSONSerialize
        ''' <summary>
        ''' Create new undefined PropertyData object
        ''' </summary>
        Public Sub New()
        End Sub

        ''' <summary>
        ''' Create and define new PropertyData object
        ''' </summary>
        ''' <param name="name">Property name</param>
        ''' <param name="text">Propery text</param>
        Public Sub New(ByVal name As String, ByVal text As String)
            Me.m_name = name
            Me.m_text = text
        End Sub

        ''' <summary>
        ''' Propery name
        ''' </summary>
        Public Property Name() As String
            Get
                Return m_name
            End Get
            Set(ByVal value As String)
                m_name = value
            End Set
        End Property

        ''' <summary>
        ''' Property text
        ''' </summary>
        Public Property Text() As String
            Get
                Return m_text
            End Get
            Set(ByVal value As String)
                m_text = value
            End Set
        End Property

        ''' <summary>
        ''' Serialize Propery data in JSON format
        ''' </summary>
        ''' <returns>JSON formated string</returns>
        Public Function JSONSerialize() As String _
            Implements IJSONSerialize.JSONSerialize
            Return ("""" & m_name & """:""") + m_text & """"
        End Function


        ''' <summary>
        ''' Protected members
        ''' </summary>
        Protected m_name As String
        Protected m_text As String
    End Class
End Namespace
